internal/exec/exec.go

```diff
@@ -85,7 +89,17 @@ func (r *Request) execSelections(ctx context.Context, sels []selected.Selection,
 				execFieldSelection(ctx, r, s, f, &pathSegment{path, f.field.Alias}, true)
 			}(f)
 		}
-		wg.Wait()
+		// close a signal channel once the wait group is complete
+		c := make(chan struct{})
+		go func() {
+			defer close(c)
+			wg.Wait()
+		}()
+		// We want to block until either the context is done or the wait group is complete
+		select {
+		case <-c:
+		case <-ctx.Done():
```

Comment:
I wonder if we need to do anything if `ctx.Error()` is non nil?

Reply:
I guess we're expecting this to only be hit when deadline is exceeded but what should we do if that's not the error?

Reply:
EDIT: looks like this is handled above when we write the final output